### PR TITLE
Add ability to listen to League events via Symfony listeners/subscribers

### PR DESCRIPTION
**Pull request by X-Coder264**
This is a very simple approach to emit League OAuth server events via the Symfony event dispatcher so that users can create their own listeners in their Symfony applications.

Closes https://github.com/thephpleague/oauth2-server-bundle/issues/35

### CODE REVIEW

File: src/Service/SymfonyEventDispatcherEmitter.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+    public function emitBatch(array $events): array
+    {
+        $results = [];
+
+        foreach ($events as $event) {
+            $results[] = $this->emit($event);
+        }
+
+        return $results;
```
Comment (ostiwe) [inline src/Service/SymfonyEventDispatcherEmitter.php:68]: Maybe use array_map ?
```suggestion
         return array_map(function ($event) {
            return $this->emit($event);
        }, $events);
```

**Comment by ajgarlag** [comment]
@X-Coder264 Instead of coding a partial implementation of `EmitterInterface` we can use a League event listener to forward league events to Symfony event dispatcher. Please, see https://github.com/X-Coder264/oauth2-server-bundle/pull/1

**Comment by X-Coder264** [comment]
@ajgarlag Thanks, I didn't know about the existence of `ListenerProviderInterface` interface. I still think this approach is better, because even though the emitter interface is bloated so you save a couple of lines with your approach, this approach ensures that all events emitted by the server will be dispatched via the Symfony dispatcher. Your approach has only some events hardcoded, which doesn't take into consideration events that can be emitted in custom grants or events that the League Server package could add in a future release (which we'd then have to manually add to that allow list afterwards in order to make them usable to the users of this bundle).

**Comment by ajgarlag** [comment]
@X-Coder264 I've seen that `league\event` supports [Wilcard Listeners](https://event.thephpleague.com/2.0/listeners/wildcard/)

**Comment by X-Coder264** [comment]
I was just looking into committing that `*` solution, you beat me to it :stuck_out_tongue:

File: docs/listening-to-league-events.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Listening to League OAuth Server events
+
+During the lifecycle of a request passing through the authorization server a number of events may be emitted.
```
Comment (chalasr) [inline docs/listening-to-league-events.md:3]:
```suggestion
During the lifecycle of a request passing through the authorization server a number of events may be dispatched.
```
just for consistency with symfony's wording

**Review by chalasr — APPROVED** [review]
Great. One minor comment, thanks!

**Comment by chalasr** [comment]
Thank you @X-Coder264.

**Pull request merged**